src/Nethermind/Nethermind.Serialization.Rlp.Benchmark/Program.cs

```diff
@@ -0,0 +1,120 @@
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Running;
+using Nethermind.Int256;
+using Nethermind.Serialization.FluentRlp;
+using Nethermind.Serialization.FluentRlp.Generator;
+
+namespace Nethermind.Serialization.Rlp.Benchmark;
+
+[RlpSerializable(RlpRepresentation.Newtype)]
+public record Address(byte[] Bytes)
+{
+    public const int Size = 20;
+}
+
+[RlpSerializable]
+public record AccessList(List<(Address, List<UInt256>)> Addresses);
+
+[MemoryDiagnoser]
+public class CurrentFluentBenchmark
+{
+    private readonly Nethermind.Core.Eip2930.AccessList _current;
+    private readonly AccessList _fluent;
+
+    public CurrentFluentBenchmark()
+    {
+        _current = Benchmark.Current.BuildAccessList(new Random(42));
+        _fluent = Benchmark.Fluent.BuildAccessList(new Random(42));
+    }
+
+    [Benchmark(Baseline = true)]
+    public Nethermind.Core.Eip2930.AccessList Current()
+    {
+        var decoder = Eip2930.AccessListDecoder.Instance;
+
+        var length = decoder.GetLength(_current, RlpBehaviors.None);
+        var stream = new RlpStream(length);
```

Comment:
This line is responsible for all the allocations as it allocates a new array underneath https://github.com/NethermindEth/nethermind/blob/4e4455960a4d62ea3429e7bb143b45789ec7246f/src/Nethermind/Nethermind.Serialization.Rlp/RlpStream.cs#L44 Is this the case that we want to benchmark or rather it should be a reused `RlpStream` here?

Reply:
To be comparable with the `FluentRlp` approach we should allocate a new buffer. Since both will allocate the same buffer size then it should not matter.

Reply:
I see. If they allocate the same buffer, what makes the current allocate over 400kb more then?

Reply:
Interestingly they're not allocating the same buffer size: the fluent approach uses a buffer of `170850` bytes while the current one uses `172845`. That does not account for the 400kb you mention though. Rider's profiler is not giving me anything useful so I'm kind of stuck now.

Maybe we should add other objects (ex. `LogEntry`, `BlockInfo`, etc.) to get more accurate benchmarks.

Reply:
you can use NettyRlpStream that uses arena memory